document-generator-core/src/test/java/uk/gov/companieshouse/document/generator/core/service/DocumentGeneratorServiceTest.java

```diff
@@ -0,0 +1,151 @@
+package uk.gov.companieshouse.document.generator.core.service;
+
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.TestInstance;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.Mock;
+import org.mockito.junit.jupiter.MockitoExtension;
+import uk.gov.companieshouse.document.generator.core.document.render.RenderDocumentRequestHandler;
+import uk.gov.companieshouse.document.generator.core.document.render.models.RenderDocumentRequest;
+import uk.gov.companieshouse.document.generator.core.document.render.models.RenderDocumentResponse;
+import uk.gov.companieshouse.document.generator.core.service.impl.DocumentGeneratorServiceImpl;
+import uk.gov.companieshouse.document.generator.core.service.models.DocumentRequest;
+import uk.gov.companieshouse.document.generator.core.service.models.DocumentResponse;
+import uk.gov.companieshouse.document.generator.interfaces.DocumentInfoService;
+import uk.gov.companieshouse.document.generator.interfaces.model.DocumentInfo;
+import uk.gov.companieshouse.environment.EnvironmentReader;
+
+import java.io.IOException;
+import java.util.HashMap;
+import java.util.Map;
+
+import static org.junit.jupiter.api.Assertions.assertNotNull;
+import static org.junit.jupiter.api.Assertions.assertNull;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.Mockito.when;
+
+@ExtendWith(MockitoExtension.class)
+@TestInstance(TestInstance.Lifecycle.PER_CLASS)
+public class DocumentGeneratorServiceTest {
+
+    @Mock
+    private DocumentInfoService mockDocumentInfoService;
+
+    @Mock
+    private RenderDocumentRequestHandler mockRequestHandler;
+
+    @Mock
+    private EnvironmentReader mockEnvironmentReader;
+
+    private DocumentGeneratorService documentGeneratorService;
+
+    @BeforeEach
+    public void setUp() {
+        documentGeneratorService = new DocumentGeneratorServiceImpl(mockDocumentInfoService,
+                mockEnvironmentReader, mockRequestHandler);
+    }
+
+    @Test
+    @DisplayName("Test a successful generate completed")
+    public void testsSuccessfulGenerateCompleted() throws IOException {
+
+        when(mockDocumentInfoService.getDocumentInfo()).thenReturn(setSuccessfulDocumentInfo());
+
+        when(mockRequestHandler.sendDataToDocumentRenderService(any(String.class), any(RenderDocumentRequest.class))).thenReturn(setSuccessfulRenderResponse());
+
+        DocumentResponse response = documentGeneratorService.generate(setValidRequest());
+
+        assertNotNull(response);
```

Comment:
Should we perform an `assertEquals` here? I'm not fussy either way, just makes the tests a little more verbose. static strings could be held at the class level to hold the DocumentInfo and RenderResponse details

Reply:
I just want to to check that the response is not null